src/resource.test.ts

```diff
@@ -0,0 +1,356 @@
+import axios from "axios";
+import MockAdapter from "axios-mock-adapter";
```

Comment:
We should just be able to use jest's built in mock here: https://jestjs.io/docs/mock-functions
They actually use `axios` as their example library